lib/feature/navigator/widgets/custom_bottom_navigation_bar.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:split/utils/locale/app_localization_keys.dart';
+
+class CustomBottomNavigationBar extends BaseStatelessWidget {
+  CustomBottomNavigationBar({Key? key}) : super(key: key);
+
+  @override
+  Widget baseBuild(BuildContext context) {
+    int currentScreen = 0;
+    return BottomAppBar(
+      shape: const CircularNotchedRectangle(),
+      child: SizedBox(
+        height: 55.h,
+        child: BlocBuilder<NavigatorBloc, NavigatorBlocSates>(
+          builder: (context, state) {
+            if (state is NavigateToGroupScreenState) {
+              currentScreen = 0;
+            } else if (state is NavigateToFriendsScreenState) {
+              currentScreen = 1;
+            } else if (state is NavigateToActivityScreenState) {
+              currentScreen = 2;
+            } else if (state is NavigateToProfileScreenState) {
+              currentScreen = 3;
+            }
+            return Row(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              children: <Widget>[
+                /// Group Icon
+                _navigationIconWidget(translate(LocalizationKeys.groups)!, 0,
+                    currentScreen, context),
+
+                /// friends Icon
+                _navigationIconWidget(translate(LocalizationKeys.friends)!, 1,
+                    currentScreen, context),
+
+                ///  activity
+                _navigationIconWidget(translate(LocalizationKeys.activity)!, 2,
+                    currentScreen, context),
+
+                // account
+                _navigationIconWidget(translate(LocalizationKeys.account)!, 3,
+                    currentScreen, context)
+              ],
+            );
+          },
+        ),
+      ),
+    );
+  }
+
+  /// ////////////////////////////////////////////////////////
+  /// ///////////////// Widget methods ///////////////////////
+  /// ////////////////////////////////////////////////////////
+  Widget _navigationIconWidget(
+      String name, int iconIndex, int currentScreen, BuildContext context) {
+    return InkWell(
+      onTap: () {
+        if (iconIndex == 0) {
+          _makeGroupsTheScreen(context);
+        } else if (iconIndex == 1) {
+          _makeFriendsTheCurrentScreen(context);
+        } else if (iconIndex == 2) {
+          _makeActivityTheCurrentScreen(context);
+        } else if (iconIndex == 3) {
+          _makeProfileTheCurrentScreen(context);
+        }
+      },
+      child: CustomNavigationBarIcon(
+          name: name, iconIndex: iconIndex, selectedIconIndex: currentScreen),
+    );
+  }
+
+  /// ////////////////////////////////////////////////////////
+  /// ///////////////// Helper methods ///////////////////////
+  /// ////////////////////////////////////////////////////////
+
+  NavigatorBloc currentBloc(BuildContext context) =>
+      context.read<NavigatorBloc>();
+
+  /// this is a list of fired events
+  _makeGroupsTheScreen(BuildContext context) {
```

Comment:
pls, decide the return type of the functions

Reply:
I made it void